frontend/common/src/components/skills/MissingSkills/MissingSkills.tsx

```diff
@@ -0,0 +1,87 @@
+import React from "react";
+import { useIntl } from "react-intl";
+
+import { getLocale } from "../../../helpers/localize";
+import type { Skill } from "../../../api/generated";
+import Chip, { Chips } from "../../Chip";
+
+export interface MissingSkillsProps {
+  requiredSkills?: Skill[];
+  optionalSkills?: Skill[];
+  addedSkills?: Skill[];
+}
+
+const getMissingSkills = (required: Skill[], added?: Skill[]) => {
+  return !added?.length
+    ? required
+    : required.filter((skill) => {
+        return !added.find((addedSkill) => addedSkill.id === skill.id);
+      });
+};
+
+const MissingSkills = ({
+  requiredSkills,
+  optionalSkills,
+  addedSkills,
+}: MissingSkillsProps) => {
+  const intl = useIntl();
+  const locale = getLocale(intl);
+
+  const missingRequiredSkills = getMissingSkills(
+    requiredSkills || [],
+    addedSkills,
+  );
+  const missingOptionalSkills = getMissingSkills(
+    optionalSkills || [],
+    addedSkills,
+  );
+
+  return (
+    <>
+      {missingRequiredSkills.length ? (
+        <>
+          <p>
+            {intl.formatMessage({
+              defaultMessage:
+                'The following "Need to have" <red>required</red> skills are missing from your profile:',
+              description: "Text",
```

Comment:
Hehe, a little more context for the translators?

Reply:
YIKES, that was me having a thought pop into my head mid typing and dropping everything before i forgot it then I forgot I was working on writing descriptions 😞 

Fixed with [provide better desc for missing skills text](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3563/commits/c89fe258a360b606b845f0cb2ecd2bc26086c4bf)